dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganizationResource.java

```diff
@@ -409,6 +410,37 @@ private Organization getOrganizationByIdOptionalAuth(Optional<User> user, Long o
         }
     }
 
+    @DELETE
+    @Path("/{organizationId}")
+    @Timed
+    @UnitOfWork
+    @ApiOperation(value = "Deletes an organization that is pending or rejected", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, hidden = true)
+    @Operation(operationId = "deleteRejectedOrPendingOrganization", summary = "Delete pending or rejected organization", description = "Delete pending or rejected organization", security = @SecurityRequirement(name = "bearer"))
+    public void deleteRejectedOrPendingOrganization(
+            @Parameter(hidden = true, name = "user") @Auth User user,
+            @Parameter(description = "Organization ID.", name = "organizationId", in = ParameterIn.PATH, required = true) @PathParam("organizationId") Long organizationId) {
+        Organization organization = organizationDAO.findById(organizationId);
+        OrganizationUser orgUser = getUserOrgRole(organization, user.getId());
+
+        // If the user does not belong to the organization or if the user is not a maintainer of the organization
+        // and if the user is neither an admin nor curator, then throw an error
+        if ((orgUser == null || orgUser.getRole() != OrganizationUser.Role.MAINTAINER) && (!user.isCurator() && !user.getIsAdmin())) {
+            throw new CustomWebApplicationException("You do not have access to delete this organization", HttpStatus.SC_FORBIDDEN);
+        }
+
+        // If the organization to be deleted is pending or has been rejected, then delete the organization
+        if (organization.getStatus() == Organization.ApplicationState.PENDING || organization.getStatus() == Organization.ApplicationState.REJECTED) {
+            organization.getStarredUsers().clear();
+            organization.getAliases().clear();
+
+            eventDAO.deleteEventByOrganizationID(organizationId);
+            organizationDAO.update(organization);
```

Comment:
Do you need this line? May be necessary because of the interaction with related entities, but it seems weird that you have to update it in order to delete it.

Not 100% sure of this, but I like it better how for users you don't have to explicitly delete (clear) them here any more. For lines 433 and 434, could you accomplish via annotations instead, and just have lines 436 and 438 in this block?

Reply:
Turns out that update() is not necessary as the session has the "up-to-date" database changes due to flush() from `eventDAO.deleteEventByOrganization`. As for lines 433 and 434, I have just removed them as it is not possible to create an alias or star an organization that has not been approved. 

I had added some new tests to ensure that the rows using the FKEYS would be deleted in organization_alias, and it turns out that the API call to get the organizationByAlias checks if the organization has been approved first. Assuming the same for starred_organizations since the feature to star will not appear until the organization has been approved. However, admins/curators can bypass the restrictions and star or add an alias to an unapproved organization through swagger (not sure if this is intended)